From ecc20cd3b4b79698fbd1caa2329d320757e00e3e Mon Sep 17 00:00:00 2001
From: "Michael S. Fischer" <mfischer@zendesk.com>
Date: Tue, 2 Jun 2015 16:29:57 -0700
Subject: [PATCH 105/120] unsquashfs: modify xattrs after other changes

Fixes a bug in which security capabilities were not properly set on
written files.  chown(2) resets all capabilities so it should be run
before setting xattrs instead of afterwards.
---
 squashfs-tools/unsquashfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/squashfs-tools/unsquashfs.c b/squashfs-tools/unsquashfs.c
index 1323dd6..a5f0117 100644
--- a/squashfs-tools/unsquashfs.c
+++ b/squashfs-tools/unsquashfs.c
@@ -821,8 +821,6 @@ int set_attributes(char *pathname, int mode, uid_t uid, gid_t guid, time_t time,
 {
 	struct utimbuf times = { time, time };
 
-	write_xattr(pathname, xattr);
-
 	if(utime(pathname, &times) == -1) {
 		ERROR("set_attributes: failed to set time on %s, because %s\n",
 			pathname, strerror(errno));
@@ -845,6 +843,8 @@ int set_attributes(char *pathname, int mode, uid_t uid, gid_t guid, time_t time,
 		return FALSE;
 	}
 
+	write_xattr(pathname, xattr);
+
 	return TRUE;
 }
 
-- 
2.8.3

